### PR TITLE
Nukestudio: Crash if save as new version with workfile - AY-854

### PR DESCRIPTION
**Pull request by tokejepsen**
## Changelog Description
Fix for edge case of saving workfile in NukeStudio that would crash the application.

## Testing notes:
1. Launch NukeStudio on task without workfiles.
2. `OpenPype -> Work Files -> Save As` and save.
3. Double click the newly created file and say 'no' to the saving question.
4. Confirm NukeStudio does not crash.

### CODE REVIEW

**Review by moonyuet — DISMISSED** [review]
When I double click the workfile and hit "no" for saving the unsaved changes, it will pop up the message to say `Failed to save workfile`
image

**Comment by tokejepsen** [comment]
@moonyuet could you post the output of the NukeStudio terminal?

I did not get this failed message, and it should still work opening the workfile.

File: client/ayon_core/hosts/hiero/api/workio.py [hunk 1]
```diff
@@ -54,10 +54,9 @@ def open_file(filepath):
     # open project file
     hiero.core.openProject(filepath.replace(os.path.sep, "/"))
 
-    # close previous project
-    project.close()
-
-
+    # Close previous project if its different to the current project.
+    if project.path() != filepath:
+        project.close()
```
[the remark this answers is not in the document]
Comment (moonyuet) [reply client/ayon_core/hosts/hiero/api/workio.py:59]: move line 55 under the condition will fix the error shown in the console below.
```
*** WRN: >>> { WorkfilesUI }: [  Open of workfile FAILED  ]
========================
Project is already open.
========================
Traceback (most recent call last):
  File "D:\ayon-core\client\ayon_core\tools\workfiles\control.py", line 512, in open_workfile
    self._open_workfile(folder_id, task_id, filepath)
  File "D:\ayon-core\client\ayon_core\tools\workfiles\control.py", line 664, in _open_workfile
    self._host_open_workfile(filepath)
  File "D:\ayon-core\client\ayon_core\tools\workfiles\control.py", line 610, in _host_open_workfile
    host.open_file(filepath)
  File "D:\ayon-core\client\ayon_core\hosts\hiero\api\workio.py", line 55, in open_file
    hiero.core.openProject(filepath.replace(os.path.sep, "/"))
RuntimeERROR: Project is already open.
```
But it is then popping up another error in the console below, of which I think is not related to this PR.
Traceback (most recent call last):
```
  File "C:\Program Files\Nuke14.0v4\pythonextensions\site-packages\hiero\core\events.py", line 65, in _callCallbacks
    callback.call(event)
  File "C:\Program Files\Nuke14.0v4\pythonextensions\site-packages\hiero\core\events.py", line 55, in call
    self.method.__call__(event, *self.args, **self.kwargs)
  File "D:\ayon-core\client\ayon_core\hosts\hiero\api\events.py", line 59, in afterProjectLoad
    add_tags_to_workfile()
  File "D:\ayon-core\client\ayon_core\hosts\hiero\api\tags.py", line 147, in add_tags_to_workfile
    task_types = project_entity["taskType"]
KeyError: 'taskType'
```

[the remark this answers is not in the document]
Comment (tokejepsen) [reply client/ayon_core/hosts/hiero/api/workio.py]: > But it is then popping up another error in the console below, of which I think is not related to this PR.
Traceback (most recent call last):

Yeah, unrelated to this PR.

File: client/ayon_core/hosts/hiero/api/workio.py [hunk 2]
```diff
@@ -51,12 +51,11 @@ def open_file(filepath):
...
-    # open project file
-    hiero.core.openProject(filepath.replace(os.path.sep, "/"))
-
-    # close previous project
-    project.close()
-
+    # Close previous project if its different to the current project.
+    if project.path() != filepath:
+        # open project file
+        hiero.core.openProject(filepath.replace(os.path.sep, "/"))
```
Comment (moonyuet) [inline client/ayon_core/hosts/hiero/api/workio.py:57]:
```suggestion
    # Close previous project if its different to the current project.
    filepath = filepath.replace(os.path.sep, "/")
    if project.path() != filepath:
        # open project file
        hiero.core.openProject(filepath)
```
I tested with multiple scenario and i find out that the code still errors out when I have unsaved changes and click the same workfile and click `No`. And the suggestion can get it fix for that scenario.

**Comment by tokejepsen** [comment]
@moonyuet it crashed on my side with your suggestion, but I think it might work for both of us now.

**Review by moonyuet — APPROVED** [review]
Tested successfully in Nuke.

**Review by mkolar — APPROVED** [review]

**Pull request merged**